src/psyclone/tests/psyir/frontend/fortran_test.py

```diff
@@ -86,46 +88,68 @@ def test_fortran_psyir_from_source():
+    psyir = fortran_reader.psyir_from_expression("-3.0 + 1.0", table)
     assert isinstance(psyir, BinaryOperation)
     assert psyir.operator == BinaryOperation.Operator.ADD
     assert isinstance(psyir.children[0], UnaryOperation)
     assert psyir.children[0].operator == UnaryOperation.Operator.MINUS
     assert isinstance(psyir.children[0].children[0], Literal)
     assert psyir.children[0].children[0].value == "3.0"
-    psyir = fortran_reader.psyir_from_expression("ABS(-3.0)")
+    psyir = fortran_reader.psyir_from_expression("ABS(-3.0)", table)
     assert isinstance(psyir, UnaryOperation)
     assert psyir.operator == UnaryOperation.Operator.ABS
     assert isinstance(psyir.children[0], UnaryOperation)
     assert psyir.children[0].operator == UnaryOperation.Operator.MINUS
     assert isinstance(psyir.children[0].children[0], Literal)
-    with pytest.raises(NotImplementedError) as err:
-        fortran_reader.psyir_from_expression("3.0 + a")
-    assert "Expression must contain only literals: '3.0 + a'" in str(err.value)
-    with pytest.raises(NotImplementedError) as err:
-        fortran_reader.psyir_from_expression("3.0 + sin(a)")
-    assert ("Expression must contain only literals: '3.0 + sin(a)'" in
+    # With kind specified by a kind parameter. Add a ContainerSymbol with a
+    # wildcard import so there's some way that 'r_def' can be brought into
+    # scope.
+    csym = table.new_symbol("kind_params_mod", symbol_type=ContainerSymbol)
+    csym.wildcard_import = True
+    fortran_reader.psyir_from_expression("3.0_r_def", table)
+    assert "r_def" in table
+    psyir = fortran_reader.psyir_from_expression("3.0_r_def", table)
+    assert isinstance(psyir, Literal)
+    assert isinstance(psyir.datatype, ScalarType)
+    assert isinstance(psyir.datatype.precision, DataSymbol)
+    assert psyir.datatype.precision is table.lookup("r_def")
+    # Symbol not found in supplied table
+    with pytest.raises(SymbolError) as err:
+        fortran_reader.psyir_from_expression("3.0 + a", SymbolTable())
+    assert ("Expression contains unresolved symbols: '3.0 + a'" in
             str(err.value))
+    # Now use the table with the container that has the wildcard import
+    psyir = fortran_reader.psyir_from_expression("3.0 + a", table)
+    assert isinstance(psyir, BinaryOperation)
+    assert "a" in table
 
 
 def test_fortran_psyir_from_expression_invalid(fortran_reader):
     ''' Test that the psyir_from_expression method raises the expected
     error when given something that is not an expression. '''
-    with pytest.raises(NotImplementedError) as err:
-        fortran_reader.psyir_from_expression("return")
-    assert "Expression must contain only literals: 'return'" in str(err.value)
+    # No supplied table
+    with pytest.raises(TypeError) as err:
+        fortran_reader.psyir_from_expression("3.0 + sin(a)", None)
+    assert ("Must be supplied with a valid SymbolTable but got 'NoneType'" in
```

Comment:
What happens if something else than `None` is supplied, e.g. another class or just some gibberish? Could you please add a test for such a case as well?

Reply:
Done.